core/vm/instructions.go

```diff
@@ -878,9 +878,11 @@ func opStop(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory
 
 func opSuicide(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
 	balance := interpreter.evm.StateDB.GetBalance(contract.Address())
-	interpreter.evm.StateDB.AddBalance(common.BigToAddress(stack.pop()), balance)
 
-	interpreter.evm.StateDB.Suicide(contract.Address())
+	suicided := interpreter.evm.StateDB.Suicide(contract.Address())
```

Comment:
Docstring for` func (self *StateDB) Suicide(addr common.Address) bool`
```
// Suicide marks the given account as suicided.
// This clears the account balance.
```
So essentially, what you do will cause instant consensus failure, since the balance will be zeroed.